xfid.go

```diff
@@ -550,32 +544,34 @@ func xfidctlwrite(x *Xfid, w *Window) {
 	// log.Println("xfidctlwrite", x)
 	// defer log.Println("done xfidctlwrite")
 	var (
-		fc              plan9.Fcall
-		err             error
-		scrdraw, settag bool
-		t               *Text
-		n               int
+		err error
+		t   *Text
 	)
-	err = nil
-	scrdraw = false
-	settag = false
+	const scrdraw = false
+	settag := false
 
 	w.tag.Commit()
 	lines := strings.Split(string(x.fcall.Data), "\n")
 	var lidx int
-	var line string
 forloop:
 	for lidx = 0; lidx < len(lines); lidx++ {
-		line = lines[lidx]
+		line := lines[lidx]
 		words := strings.SplitN(line, " ", 2)
 		switch words[0] {
 		case "": // empty line.
-		case "lock": // make window exclusive use
-			w.ctrllock.Lock()
-			w.ctlfid = x.f.fid
-		case "unlock": // release exclusive use
-			w.ctlfid = math.MaxUint32
-			w.ctrllock.Unlock()
+
+		// Lock/unlock can hang or crash Edwood.
+		// They don't appear to be used for anything useful, so disable for now.
+		//
+		//case "lock": // make window exclusive use
```

Comment:
maybe print something diagnostic about this in case something stops working?

Reply:
I guess I can add a print. I don't know of anything that uses it, but this might be a way to find out. Usually clients are too lazy to check for write errors, so they wouldn't report about the failed write.

Reply:
Done

Reply:
Thanks.